src/Highway.Data.EntityFramework/Interfaces/IReadonlyEntityDataContext.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+using System.Data.Common;
+
+namespace Highway.Data.EntityFramework
+{
+    public interface IReadonlyEntityDataContext : IReadonlyDataContext
+    {
+        /// <summary>
+        ///     Executes a SQL command and tries to map the returned datasets into an <see cref="IEnumerable{T}" />
+        ///     The results should have the same column names as the Entity Type has properties
+        /// </summary>
+        /// <typeparam name="T">The Entity Type that the return should be mapped to</typeparam>
+        /// <param name="sql">The Sql Statement</param>
+        /// <param name="dbParams">A List of Database Parameters for the Query</param>
+        /// <returns>An <see cref="IEnumerable{T}" /> from the query return</returns>
+        IEnumerable<T> ExecuteSqlQuery<T>(string sql, params DbParameter[] dbParams);
```

Comment:
It could be argued that moving this method to the read-only interface isn't safe, if it is just a SQL string and params. But the _intent_ of the name "ExecuteSqlQuery" seems obvious. Thoughts?

Reply:
I agree that it is dangerous, but for complex queries where a stored proc are the right answer this is a lot of give up. Ultimately this is only accessible through advanced queries, so my gut is to move it and keep that level of visibility. A warning in the description that it can break the read-only nature if use improperly?

Reply:
@DevlinLiles another option would be to append ApplicationIntent=Readonly to the connection string, but this won't work the way I did it with the DataContext inheriting from ReadOnlyDataContext.  Your "configuration vs. two data context types" suggestion below would allow it, though.  See additional comments on the ReadonlyDataContext type.